### PR TITLE
Add bike shop

### PR DESCRIPTION
**Pull request by nornagon**
From my mod: https://github.com/nornagon/cdda-more-city-locations

I'm unsure about the spawn rates here, let me know if they need adjusting :)

### CODE REVIEW

File: data/json/mapgen/bike_shop.json [hunk 1]
```diff
@@ -0,0 +1,159 @@
+[
+  {
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": [ "s_bike_shop" ],
+    "object": {
+      "fill_ter": "t_grass",
```
Comment (Coolthulhu) [inline data/json/mapgen/bike_shop.json]: If you don't need to fill, it's better if you don't. That way the mapgen entry can be modernized easier in the future.

File: data/json/mapgen/bike_shop.json [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      "place_vehicles": [
+        { "vehicle": "bikeshop", "x": [ 4, 6 ], "y": [ 3, 4 ], "chance": 90, "rotation": 270 },
+        { "vehicle": "bikeshop", "x": [ 7, 10 ], "y": 4, "chance": 90, "rotation": 270 },
+        { "vehicle": "bikeshop", "x": [ 11, 14 ], "y": 4, "chance": 90, "rotation": 270 },
+        { "vehicle": "bikeshop", "x": 12, "y": 8, "chance": 100, "rotation": 270, "status": 0 },
+        { "vehicle": "bikeshop", "x": 19, "y": 8, "chance": 100, "rotation": 270, "status": 0 },
+        { "vehicle": "bikeshop", "x": [ 5, 6 ], "y": [ 8, 9 ], "chance": 50, "rotation": 0 },
+        { "vehicle": "bikeshop", "x": [ 5, 6 ], "y": [ 10, 11 ], "chance": 50, "rotation": 0 },
+        { "vehicle": "bikeshop", "x": [ 5, 7 ], "y": [ 12, 13 ], "chance": 50, "rotation": 0 },
+        { "vehicle": "bikeshop", "x": [ 5, 7 ], "y": [ 14, 15 ], "chance": 50, "rotation": 0 }
```
Comment (Coolthulhu) [inline data/json/mapgen/bike_shop.json]: Don't specify 0 rotation. It is the default.

File: data/json/mapgen/bike_shop.json [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        { "group": "bikeshop_tools", "chance": 80, "repeat": [ 1, 2 ], "x": [ 10, 10 ], "y": [ 12, 13 ] },
+        { "group": "bikeshop_for_sale", "chance": 100, "repeat": [ 1, 2 ], "x": [ 16, 18 ], "y": 18 },
+        { "group": "bikeshop_for_sale", "chance": 100, "repeat": [ 1, 3 ], "x": 19, "y": [ 13, 16 ] }
+      ],
+      "place_vehicles": [
+        { "vehicle": "bikeshop", "x": [ 4, 6 ], "y": [ 3, 4 ], "chance": 90, "rotation": 270 },
+        { "vehicle": "bikeshop", "x": [ 7, 10 ], "y": 4, "chance": 90, "rotation": 270 },
+        { "vehicle": "bikeshop", "x": [ 11, 14 ], "y": 4, "chance": 90, "rotation": 270 },
+        { "vehicle": "bikeshop", "x": 12, "y": 8, "chance": 100, "rotation": 270, "status": 0 },
+        { "vehicle": "bikeshop", "x": 19, "y": 8, "chance": 100, "rotation": 270, "status": 0 },
```
Comment (Coolthulhu) [inline data/json/mapgen/bike_shop.json]: Status 0 means pristine. They probably shouldn't have 100% chance to spawn.

**Comment by Coolthulhu** [comment]
Lacks an overmap terrain entry.
When adding it, use the `"copy-from"`, like other shops.

File: data/json/mapgen/bike_shop.json (hunk 2 above)
Comment (nornagon) [reply data/json/mapgen/bike_shop.json]: Done.

File: data/json/mapgen/bike_shop.json (hunk 3 above)
Comment (nornagon) [reply data/json/mapgen/bike_shop.json]: Toned down to 60%.

File: data/json/mapgen/bike_shop.json (hunk 1 above)
Comment (nornagon) [reply data/json/mapgen/bike_shop.json]: Done.

**Comment by nornagon** [comment]
overmap_terrain entry added.

**Comment by nornagon** [comment]
Also reduced the frequency in generation from 2 to 1 to balance out some of the sweet loot you might find, and because bike shops in rural New England don't seem that common.

**Comment by Coolthulhu** [comment]
Small merge conflicts.
Also, I'm still getting errors on load.

**Comment by nornagon** [comment]
Merge conflicts resolved, load error fixed.

**Comment by nornagon** [comment]
Ping @Coolthulhu, is this good to merge now?

**Comment by kevingranade** [comment]
The bikes sometimes generate smashed into the walls, be careful of that.
I'm fixing it up while merging.
You can do debug -> g -> o <choose your building type> <enter> <enter> <enter> ... to regenerate a bunch of times and check for stuff like this.

**Pull request merged**